### ISSUE TITLE
questions and potential suggestions

### ISSUE THREAD

**Issue by Darkmega18** [trentv4/AdvancedDarkness#2]
G'day, I was wondering if theres any way that this can be enforced only on certain dimensions. I was hoping to keep my overworld and base dimensions happy but then darken dimensions like deep dark or my "darker more often than normal" overworld clone and such.

Also is there thought of making this into an RPG like hazard with things like mobs, gameplay mechanics or structures/blocks for dispersing/causing darkness? Something akin to having dimensions with advanced darkness spawn mobs at closer ranges to the player if the block is full dark or perhaps something like metroid prime 2 echoes with the Ing and light shards/pillars spawning in the world that can make bigger light or disperse the darkness effects in an area like the comfort lanterns?

**Comment by trentv4** [comment]
I split your request into #3 for per-dimension darkness just so it's easier to manage.

I do not intend on adding more mobs or hazards. I think this mod should be able to be used as a strictly utility mod without any world generation or outside effects... however, I am interested in what you mean by the "echo with the ing" and "light shards/pillars spawning in the world". Can you elaborate? I'm not a metroid player.

**Comment by Darkmega18** [comment]
well in metroid prime 2 echoes there was essentially 2 worlds that samus goes between. Aether and then a dark aether. Look up dark aether for details and such so I don't go off on a tangent, but essentially dark aethers environment is toxic by default to samus and just walking around in it causes damage over time.

But there are crystals that essentially work as safe spots which produce an atmosphere/light bubble when you shoot them with stops DoT and actually verily slowly regenerates health. It also has the added effect of flicking the Ing, which are the dark creatures which are part of dark aether. Later on when you get the light beam you can also overcharge them to create a hardened light shield that just kills enemies when they try to encroach through it at you, while some enemy's attacks and samuses dark beam can instead nullify the crystals with darkness to stop them from working.

So in applications to this mod for areas where if there was super darkness these posts with light crystals or light orbs that produce light bubbles when attacked/punched/activated would essentially be zones that allow the player to see normally without extreme darkness fogging the area, flick mobs and produce light in an extended area, something akin to a thaumcraft arcane lamp if it still has the effect of producing light clouds in a wide area. These would either be crafted by the player, or spawn in worlds that are set to contain darkness on a per chunk basis as safezones for players without access to making their own yet.

**Comment by trentv4** [comment]
See 9ed759e and #3. I added the additional configurations you wanted.

I do not think I will be adding anything to do with world generation. I like the idea, but I believe advanced darkness should remain a utility mod that can just be dropped in and not mess with world generation. The only way I would do the generation you describe is if I added a new dimension to put it in. I may do that later, but not now.
